### ISSUE TITLE
Template Class that inherits from GP

### ISSUE THREAD

**Issue by cjk12** [convexopt/gpkit#159]
Discussed with @bqpd , would be nice to implement.

Would facilitate post solve operations.

**Comment by cjk12** [comment]
I'm working on this for the Rotor and would like to do the following:

```python
import numpy as np
from gpkit import GP

class Rotor(GP):
    def __init__(self, cost, constraints, substitutions={}, solver=None, options={}):
        GP.__init__(self, cost, constraints, substitutions={}, solver=None, options={})

    def calc(self):
        self.y               = self.r*self.R   
```

Problem is it is failing, because I actually want calc to operate on the `gpkit.geometric_program.GPSolutionArray` not the Rotor class that is equivalent to `GP`.  Any thoughts on how best to implement?

**Comment by cjk12** [comment]
Lines of code similar to this:

```python
from ._mosek import cli_expopt
```

Are really starting to hurt.  Can we investigate more rigorously defining the places we are importing from so that references work for subclasses?

**Comment by cjk12** [comment]
Lines such as these are also giving me some trouble:

```python
solution.append(self.__run_solver())
```

because this function is not defined in self, but in the original class.  Again, possibly an area to be investigated?

**Comment by bqpd** [comment]
Implemented in 5bb263212f9143718b63729df372f7eea8a4dc71
